presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -94,395 +117,640 @@
 
-            Scope scope = Scope.builder()
-                    .withRelationType(RelationId.anonymous(), new RelationType(fields.build()))
-                    .build();
-
-            // Given base query's filter condition and materialized view's filter condition, the goal is to check if materialized view's
-            // filters contain Base's filters (Base implies materialized view).
-            // Let base query's filter condition be A, and materialized view's filter condition be B.
-            // One way to evaluate A implies B is to evaluate logical expression A^~B and check if the output domain is none.
-            // If the resulting domain is none, then A^~B is false. Thus A implies B.
-            // For more information and examples: https://fb.quip.com/WwmxA40jLMxR
-            // TODO: Implement method that utilizes external SAT solver libraries. https://github.com/prestodb/presto/issues/16536
-            RowExpression materializedViewWhereCondition = convertToRowExpression(materializedViewInfo.getWhereClause().get(), scope);
-            RowExpression baseQueryWhereCondition = convertToRowExpression(node.getWhere().get(), scope);
-            RowExpression rewriteLogicExpression = and(baseQueryWhereCondition,
-                    call(baseQueryWhereCondition.getSourceLocation(),
-                            "not",
-                            new FunctionResolution(metadata.getFunctionAndTypeManager()).notFunction(),
-                            materializedViewWhereCondition.getType(),
-                            materializedViewWhereCondition));
-            RowExpression disjunctiveNormalForm = logicalRowExpressions.convertToDisjunctiveNormalForm(rewriteLogicExpression);
-            ExtractionResult<VariableReferenceExpression> result = domainTranslator.fromPredicate(session.toConnectorSession(), disjunctiveNormalForm, BASIC_COLUMN_EXTRACTOR);
-
-            if (!result.getTupleDomain().equals(TupleDomain.none())) {
-                throw new IllegalStateException("View filter condition does not contain base query's filter condition");
-            }
+    @Override
+    protected Node visitIntersect(Intersect node, Void context)
+    {
+        List<Relation> newRelations = processNodes(node.getRelations());
+        if (node.getRelations() == newRelations) {
+            return node;
         }
 
-        return new QuerySpecification(
-                (Select) process(node.getSelect(), context),
-                node.getFrom().map(from -> (Relation) process(from, context)),
-                node.getWhere().map(where -> (Expression) process(where, context)),
-                node.getGroupBy().map(groupBy -> (GroupBy) process(groupBy, context)),
-                node.getHaving().map(having -> (Expression) process(having, context)),
-                node.getOrderBy().map(orderBy -> (OrderBy) process(orderBy, context)),
-                node.getOffset(),
-                node.getLimit());
+        return new Intersect(
```

Comment:
We should only break it into multiple lines if it does not fit in single line. Similar for other places.